cardinal/system/system_attack.go

```diff
@@ -5,57 +5,60 @@ import (
 
 	comp "github.com/argus-labs/starter-game-template/cardinal/component"
 	"github.com/argus-labs/starter-game-template/cardinal/tx"
-	"pkg.world.dev/world-engine/cardinal/ecs"
-	"pkg.world.dev/world-engine/cardinal/ecs/entity"
-	"pkg.world.dev/world-engine/cardinal/ecs/filter"
-	"pkg.world.dev/world-engine/cardinal/ecs/log"
-	"pkg.world.dev/world-engine/cardinal/ecs/transaction"
+	"pkg.world.dev/world-engine/cardinal"
 )
 
 // AttackSystem is a system that inflict damage to player's HP based on `AttackPlayer` transactions.
 // This provides a simple example of how to create a system that modifies the component of an entity.
-func AttackSystem(world *ecs.World, tq *transaction.TxQueue, _ *log.Logger) error {
+func AttackSystem(wCtx cardinal.WorldContext) error {
 	// Get all the transactions that are of type CreatePlayer from the tx queue
-	attackTxs := tx.AttackPlayer.In(tq)
+	attackTxs := tx.AttackPlayer.In(wCtx)
 
 	// Create an index of player tags to its health component
-	playerTagToID := map[string]entity.ID{}
-	ecs.NewQuery(filter.Exact(comp.Player, comp.Health)).Each(world, func(id entity.ID) bool {
-		player, err := comp.Player.Get(world, id)
+	playerTagToID := map[string]cardinal.EntityID{}
+	q, err := wCtx.NewSearch(cardinal.Exact(comp.PlayerComponent{}, comp.HealthComponent{}))
+	if err != nil {
+		return err
+	}
+	err = q.Each(wCtx, func(id cardinal.EntityID) bool {
+		player, err := cardinal.GetComponent[comp.PlayerComponent](wCtx, id)
 		if err != nil {
 			return true
 		}
 
 		playerTagToID[player.Nickname] = id
 		return true
 	})
+	if err != nil {
+		return err
+	}
 
 	// Iterate through all transactions and process them individually.
 	// DEV: it's important here that you don't break out of the loop or return an error here
 	// or otherwise the rest of the transaction will not be processed & get dropped.
 	// In the future, you will be able to add error receipts to transaction receipts.
 	for _, attack := range attackTxs {
```

Comment:
We should use the "TxIterator" functionality that @jerargus built for all the systems that read a TxQueue, there is an example usage of `ForEach` in the test [here](https://github.com/Argus-Labs/world-engine/pull/303/files)